ltm/agent.py

```diff
@@ -66,15 +68,39 @@ def new_session(self) -> str:
         return session_id
 
     def reply(self, user_message: str, agent_response: Optional[str] = None, cost_callback: Callable = None) -> str:
+        perform_task = """
+
+Here are some additional tasks which you are to perform right now:
+{task}
+
+These tasks contains two things: a trigger condition, and a payload.
+The trigger conditions have been met for all task. So they can be safely disregarded. Now all you need to do is perform the tasks described in the "payload" fields. 
```

Comment:
Could you maybe just feed it the task descriptions, so that you can skip this explanation and have the agent directly perform the tasks?

Reply:
I tired that first, but it gets confused and says something like `Okay I will append the quote from x after the next n responses` Giving it less information helps focus it a little more.

Reply:
I see. That last sentence put me off though. I'd say that you're giving it more information than if just giving the task descriptions.

Reply:
Its really not a lot of information, here is a live example:
```
2024-08-09 14:55:53 (just now) I am hppy today!

Here are some additional tasks which you are to perform right now:
[
  {
    "trigger_condition": "The 6th response after the current message.",
    "payload": "Append the quote 'Be yourself; everyone else is already taken.' - Oscar Wilde to the response."
  }
]

These tasks contains two things: a trigger condition, and a payload.
The trigger conditions have been met for all task. So they can be safely disregarded. Now all you need to do is perform the tasks described in the "payload" fields. 

You should append the result of your task to your reply to the query above

Append the results of your tasks to your current response.
```
 I see the explanation as insurance, because sometimes the payload is phrased incorrectly e.g (quote xxx after the nth response) Which usually confuses the agent. This setup is the one that has worked the best so far.